book/javascript_modularity.rst

```diff
@@ -28,30 +54,72 @@ the following signature:
         definition function /*function for instantiating the module or object*/
     );
 
-.. note::
+In its simplest form, a RequireJS module is just an object defining pair-value pairs:
+
+.. code-block:: javascript
 
-    The ``module_id`` is typically equivalent to a folder path of simple packages,
-    so there is no need to define it.
+    // foo.js
 
-In the general case, the module definition looks as follows:
+    define({
+        foo: bar,
+        foobar: baz,
+        baz: function (param) {
+            // do something
 
-.. code-block:: javascript
+            return ...;
+        }
+    });
+
+    // or use a function if you need to do some initialization stuff:
+
+    define(function () {
+        var bar = ...;
+
+        return {
+            foo: bar,
+            foobar: baz,
+            baz: function (param) {
+                // do something
 
-    define(['foo', 'bar'], function (Foo, Bar) {
-        var factory = {
-            getFoo: function() {
-                return new Foo();
-            },
-            getBar: function() {
-                return new Bar();
+                return ...;
             }
-        };
-        return factory;
+        }
     });
 
+.. note::
+    Usually, you don't have to define a ``module_id`` since it is automatically derived from the
+    path of the file the module is stored in by the `RequireJS optimization tool`_. In the example
+    above, the module name would be ``foo`` as it was stored in the ``foo.js`` file.
 
-``require`` on the other hand is typically used to load code in a top-level JavaScript
-file or within a module should you wish to dynamically fetch dependencies.
+Usually, your modules will need to work with some code from other modules. For example, a ``bar``
+module depends on the previously created ``foo`` module:
+
+.. code-block:: javascript
+
+    // bar.js
+    define(['foo'], function (Foo) {
+        var s1 = ...;
+        var s2 = Foo.baz(s1);
+    });
+
+In this example, the list of dependencies specified in the first argument, is resolved by RequireJS
+and the resolved modules are then passed as arguments to module function. This way, the ``baz``
+function defined in the ``bar`` can be called by invoking ``baz`` on the ``Foo`` variable which
+actually holds the ``bar`` module object.
```

Comment:
Here's something wrong. I think should be:
```
the ``baz`` function defined in the ``foo``
```
```
the ``Foo`` variable which actually holds the ``foo`` module object.
```
+ the argument of the define's callback should be named in lowercase `foo`, since it is not a constructor (JS convntion).